qualityassurance/qastagesformattingguidelines.md

```diff
@@ -0,0 +1,88 @@
+# ♻️ QA Testing Stages, Formatting and GuideLines
+
+This Section is to instruct on when to test, how you should be formatting the Bugs raised and GuideLines for QA
+
+## Developing with QA in mind
+
+### Commands
+When you are developing the project the project might require commands to run and work correctly, these can range from simple set up commands to automatic updating commands, a list of these commands and a breif description of what these commands do will help QA move more smoothly and avoid raising unnecessary questions.
+
+Some projects requires certain status or manipulation of the database to update systems and cause it to do the next stage/part of the system to test. We would like these to be formed into Commands, QA should not need to touch the database side to test the system, this leads to user error and this is leading it away from how the customer would user the system. So these processes need to put into commands where it will update the database for the required data with the correct process to ensure that it is changed to how the system will change it when on live.
```

Comment:
I disagree with this line - Spending extra development time to write commands to automatically trigger stages outside of the lifecycle of the application I don't feel is effective use of dev time, QA should typically move through the application as a user so commands should usually never be necessary to trigger.

For testing that does require certain conditions to be met like a date being past a certain other date etc, changing the value in the database is easy enough & IMO doesn't warrant dedicated development time.

I do agree commands should be written for very complex processes & triggers that wouldn't be practical to achieve by the tester but these are far and few between, there are a lot more cost effective solutions though like manually triggering Chron jobs etc.

In relation to QA not having to touch the DB, it depends on the kind of testing taking place. black box testing needs access to the database. white box testing does not etc.

@MaxHarrisonGit I'd love to hear your opinions on this.

Reply:
Thanks for the feedback, As we are doing black box and white box testing we would need the option to do it entirely from the front end with only the bare minimum manipulation of data to test, to ensure that everything it working when you don't manipulate the data. 
This was more in general, Because we want to do box White and Black box testing we need to have both taken account for when developing. 

In terms of doing simple commands these are "Required" however it would have QA if we have new users come in, seeing a large database and being told to just change 2 things in two what seems to be random tables, you can easily forget, so either having a nice simple command set there, or having documentation of what is exactly need to do and what the outcome is and why would would do it, either would help flow and help improve the speed of testing. 

@AlexCatch Would Like to hear your feedback to this 😄

Reply:
I'm happy to require documentation for procedures of steps and triggers that aren't obvious & commands for edge cases 👍

Reply:
@AlexCatch I have edited to Document to Reflect this 😄Can give it a quick read over and confirm that We are on the same page